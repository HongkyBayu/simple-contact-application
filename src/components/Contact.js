import React from 'react';
import { View, Text, Image, StyleSheet } from 'react-native';

export default function Contact({ contactName, contactEmail , imageURL}) {
  return (
      <View style={styles.contactContainer}>
        <Image style={{ width: 60, height: 60 }} source={{ uri: imageURL }}/>
        <Text style={styles.name}>{contactName}</Text>
        <Text style={styles.email}>{contactEmail}</Text>
      </View>
  );
}

const styles = StyleSheet.create({
  contactContainer: {
    flex: 1,
    alignItems: 'flex-end',
    flexDirection: 'row',
    borderWidth: 0.5,
  },
  name: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },
  email: {
    fontSize: 10,
    padding: 5,
    height: 30,
  },
});