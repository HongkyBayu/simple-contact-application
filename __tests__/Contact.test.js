import 'react-native';
import { mount } from 'enzyme';
import React from 'react';
import Contact from '../src/components/Contact';
import renderer from 'react-test-renderer';

describe('Contact', () => {
  describe('#showData', () => {
    it('should render the matching snapshot correctly', () => {
      const app = renderer.create(<Contact/>).toJSON();
      expect(app).toMatchSnapshot();
    });

    it('should show data from the props name chetan email chetan@mail.com image chetan.png', () => {
      const wrapper = mount(<Contact contactName={'chetan'} contactEmail={'chetan@mail.com'} imageURL={'chetan.png'}/>)
      const nameContent = wrapper.find('Text').at(1);
      const emailContent = wrapper.find('Text').at(2);
      expect(nameContent.text()).toEqual('chetan');
      expect(emailContent.text()).toEqual('chetan@mail.com');
    });

    it('should show data from the props name neel email neel@mail.com image chetan.png', () => {
      const wrapper = mount(<Contact contactName={'neel'} contactEmail={'neel@mail.com'} imageURL={'chetan.png'}/>)
      const nameContent = wrapper.find('Text').at(1);
      const emailContent = wrapper.find('Text').at(2);
      expect(nameContent.text()).toEqual('neel');
      expect(emailContent.text()).toEqual('neel@mail.com');
    });
  });
});