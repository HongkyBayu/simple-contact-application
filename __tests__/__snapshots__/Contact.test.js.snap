// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact #showData should render the matching snapshot correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "borderWidth": 0.5,
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "height": 60,
        "width": 60,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 18,
        "height": 44,
        "padding": 10,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 10,
        "height": 30,
        "padding": 5,
      }
    }
  />
</View>
`;
